addon/adapter.go

```diff
@@ -80,7 +80,9 @@ func (h *Adapter) Run(addon func() error) {
 			if _, soft := err.(interface{ Soft() *SoftError }); !soft {
 				Log.Error(err, "Addon failed.")
 			}
-			h.Failed(err.Error())
+			if h.client.Error == nil {
+				h.Failed(err.Error())
```

Comment:
**hint:** Don't report the error with the client (to the hub) when the client has an error because the client cannot talk to the hub.